src/state/utils.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  IElementDefinition,
+  IElementDefinition_Type
+} from "@ahryman40k/ts-fhir-types/lib/R4";
+import { AxiosResponse } from "axios";
+import {
+  FetchedDataCodeSystem,
+  fhirDataState,
+  SimplifiedAttributes,
+  RenderAttributesTree
+} from "types";
+
+export const renderTreeAttributes = (
+  attribute: SimplifiedAttributes,
+  rootPath: SimplifiedAttributes,
+  node: RenderAttributesTree,
+  rootNode: RenderAttributesTree
+): RenderAttributesTree => {
+  if (node.id === rootPath.path) {
+    return rootNode;
+  } else {
+    const splitPath = attribute.path.split(".");
+    const childNode = node.children.find((c) => c.name === splitPath[0]);
+    const newPath = splitPath.slice(1, splitPath.length).join(".");
+    const newAttribute = {
+      path: newPath,
+      type: attribute.type,
+      definition: attribute.definition,
+      min: attribute.min,
+      max: attribute.max
+    };
+    if (childNode) {
+      return renderTreeAttributes(newAttribute, rootPath, childNode, rootNode);
+    } else {
+      const newNode = rootPath.valueSet
+        ? {
+            name: splitPath[0],
+            id: rootPath.path,
+            type: rootPath.type,
+            children: [],
+            definition: rootPath.definition,
+            valueSet: rootPath.valueSet,
+            min: rootPath.min,
+            max: rootPath.max
+          }
+        : {
+            name: splitPath[0],
+            id: rootPath.path,
+            type: rootPath.type,
+            definition: rootPath.definition,
+            children: [],
+            min: rootPath.min,
+            max: rootPath.max
+          };
```

Comment:
```suggestion
      const newNode = {
            name: splitPath[0],
            id: rootPath.path,
            type: rootPath.type,
            children: [],
            definition: rootPath.definition,
            valueSet: rootPath.valueSet,
            min: rootPath.min,
            max: rootPath.max
          };
```
Why doesn't this work ?